pkg/proxy/request/resources.go

```diff
@@ -34,6 +35,7 @@ import (
 // Resources is a collection of resources a Trickster request would need to fulfill the client request
 // This is stored in the client request's context for use by request handers.
 type Resources struct {
+	sync.Mutex
```

Comment:
I don't like exposing this mutex.

should I add getter/setters for the `rsc.TimeRangeQuery, rsc.TSReqestOptions` fields?

Reply:
I think it is fine either way. Embedding a Mutex like this is pretty common / accepted and does not concern me.